pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SparkRBackend.scala

```diff
@@ -0,0 +1,99 @@
+package edu.berkeley.cs.amplab.sparkr
+
+import java.io.IOException
+import java.net.InetSocketAddress
+import java.util.concurrent.TimeUnit
+
+import io.netty.bootstrap.ServerBootstrap
+import io.netty.channel.ChannelFuture
+import io.netty.channel.ChannelInitializer
+import io.netty.channel.ChannelOption
+import io.netty.channel.EventLoopGroup
+import io.netty.channel.nio.NioEventLoopGroup
+import io.netty.channel.socket.SocketChannel
+import io.netty.channel.socket.nio.NioServerSocketChannel
+
+import io.netty.handler.codec.bytes.ByteArrayDecoder
+import io.netty.handler.codec.bytes.ByteArrayEncoder
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder
+
+/**
+ * Netty-based backend server that is used to communicate between R and Java.
+ */
+class SparkRBackend() {
+
+  var channelFuture: ChannelFuture = null  
+  var bootstrap: ServerBootstrap = null
+  var bossGroup: EventLoopGroup = null
+
+  def init(port: Int) {
+    bossGroup = new NioEventLoopGroup(SparkRConf.numServerThreads)
+    val workerGroup = bossGroup
+    val handler = new SparkRBackendHandler(this)
+  
+    bootstrap = new ServerBootstrap()
+      .group(bossGroup, workerGroup)
+      .channel(classOf[NioServerSocketChannel])
+  
+    bootstrap.childHandler(new ChannelInitializer[SocketChannel]() {
+      def initChannel(ch: SocketChannel) = {
+        ch.pipeline()
+          .addLast("encoder", new ByteArrayEncoder())
+          .addLast("frameDecoder",
+            // maxFrameLength = 2G
+            // lengthFieldOffset = 0
+            // lengthFieldLength = 4
+            // lengthAdjustment = 0
+            // initialBytesToStrip = 4, i.e. strip out the length field itself
+            new LengthFieldBasedFrameDecoder(Integer.MAX_VALUE, 0, 4, 0, 4))
+          .addLast("decoder", new ByteArrayDecoder())
+          .addLast("handler", handler)
+      }
+    })
+
+    channelFuture = bootstrap.bind(new InetSocketAddress(port))
```

Comment:
Create the address at the beginning of the function? If any issues occur we'd catch it earlier.